stairway/src/main/java/bio/terra/stairway/FlightFilter.java

```diff
@@ -23,18 +26,42 @@
  * That filter would return ingest flights completed in the last day (assuming you'd properly set
  * the value of {@code yesterday}) run by user ddtest (assuming you'd passed the email as an input
  * parameter.
+ *
+ * <p>You can also initialize the filter with more complex boolean filters:
+ *
+ * <pre>{@code
+ * FlightFilter filter = new FlightFilter(
+ *     makeAnd(
+ *         makePredicateFlightStatus(EQUAL, FlightStatus.ERROR),
+ *         makePredicateInput("other_num", LESS_THAN, 456)),
+ *         makeOr(
+ *             makePredicateInput("email", EQUAL, "ddtest@gmail.com"),
+ *             makePredicateInput("num", GREATER_THAN, 123)))
+ *     .submittedTimeSortDirection(FlightFilterSortDirection.DESC)
+ * }</pre>
  */
 public class FlightFilter {
   private final List<FlightFilterPredicate> flightPredicates;
   private final List<FlightFilterPredicate> inputPredicates;
+  private FlightBooleanOperationExpression inputBooleanOperationExpression;
   private FlightFilterSortDirection submittedTimeSortDirection;
-  private int parameterId;
+
+  // Mapper should be used to deserialize to generic Postgres JSON
+  private static final ObjectMapper pgJsonMapper = new ObjectMapper();
 
   public FlightFilter() {
     flightPredicates = new ArrayList<>();
     inputPredicates = new ArrayList<>();
     submittedTimeSortDirection = FlightFilterSortDirection.ASC;
-    parameterId = 0;
+  }
+
+  /**
+   * Use this constructor to allow for more complex selection criteria. This takes in expression
+   * builder methods that are created using static methods on this class.
+   */
+  public FlightFilter(FlightBooleanOperationExpression inputBooleanOperationExpression) {
+    this();
```

Comment:
In general I think it's better to chain from fewer params to more, so it's explicit that `inputBooleanOperationExpression` is `null` in one case. Then you could make `inputBooleanOperationExpression` `final`.

You could also also change the other fields to be initialized at their declarations, but it could work either way.
```
public class FlightFilter {
  private final List<FlightFilterPredicate> flightPredicates = new ArrayList<>();
  private final List<FlightFilterPredicate> inputPredicates = new ArrayList<>();
  private final FlightBooleanOperationExpression inputBooleanOperationExpression;
  private FlightFilterSortDirection submittedTimeSortDirection = FlightFilterSortDirection.ASC;

  // Mapper should be used to deserialize to generic Postgres JSON
  private static final ObjectMapper pgJsonMapper = new ObjectMapper();

  public FlightFilter() {
    this(null);
  }

  /**
   * Use this constructor to allow for more complex selection criteria. This takes in expression
   * builder methods that are created using static methods on this class.
   */
  public FlightFilter(FlightBooleanOperationExpression inputBooleanOperationExpression) {
    this.inputBooleanOperationExpression = inputBooleanOperationExpression;
  }
```

Reply:
ah good call!